cli/impl/src/main/java/io/helidon/build/cli/impl/InitCommand.java

```diff
@@ -205,8 +205,15 @@ private boolean isSupportedVersion(MavenVersion version, boolean notFoundWillFai
         return false;
     }
 
-    private void assertSupportedVersion(String helidonVersion) {
-        MavenVersion version = MavenVersion.toMavenVersion(helidonVersion);
+    private void assertSupportedVersion(String helidonVersion) throws Metadata.UpdateFailed {
```

Comment:
Does the init command need to resolve the version ? E.g. convert `3-LATEST` to `3.2.1`.
If the version string contains a wildcard it will be used as is to lookup the archetype file right ?

Reply:
Where it should be in that case?

Reply:
The method `assertSupportedVersion` does not make sense anymore because the version can contain a wildcard and thus we need to resolve it.

You can use something like this instead:

```java
private String resolveHelidonVersion(String helidonVersion) throws Metadata.UpdateFailed {
    ArchetypesData data = metadata().archetypesData();
    MavenVersion resolved = VersionRange.wildcard(helidonVersion).matchVersion(data.mavenVersions());
    if (resolved != null) {
        return resolved.toString()
    }
    failed(VERSION_NOT_FOUND_MESSAGE);
}
```

The usage can be something like this:

```java
String helidonVersion = initOptions.helidonVersion();
if (helidonVersion == null) {
    ArchetypesData archetypesData = metadata.archetypesData();
    if (context.properties().containsKey(HELIDON_VERSION_PROPERTY)) {
        helidonVersion = resolveVersion(context.properties().getProperty(HELIDON_VERSION_PROPERTY));
    } else if (batch) {
        helidonVersion = archetypesData.defaultVersion();
        Log.info("Using Helidon version " + helidonVersion);
    } else {
        helidonVersion = promptHelidonVersion(archetypesData, true);
    }
} else {
    helidonVersion = resolveVersion(helidonVersion);
}
initOptions.helidonVersion(helidonVersion); // This is what the invoker uses to lookup the archetype on the file system
```